src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -2665,12 +2681,13 @@ private void connectHelper(ServerPortPlaceHolder serverInfo, int timeOutSliceInM
 
         // if the timeout is infinite slices are infinite too.
         tdsChannel = new TDSChannel(this);
+        InetSocketAddress inetSocketAddress;
         if (0 == timeOutFullInSeconds)
-            tdsChannel.open(serverInfo.getServerName(), serverInfo.getPortNumber(), 0, useParallel, useTnir,
-                    isTnirFirstAttempt, timeOutsliceInMillisForFullTimeout);
-        else
-            tdsChannel.open(serverInfo.getServerName(), serverInfo.getPortNumber(), timeOutSliceInMillis, useParallel,
+            inetSocketAddress = tdsChannel.open(serverInfo.getServerName(), serverInfo.getPortNumber(), 0, useParallel,
                     useTnir, isTnirFirstAttempt, timeOutsliceInMillisForFullTimeout);
+        else
+            inetSocketAddress = tdsChannel.open(serverInfo.getServerName(), serverInfo.getPortNumber(),
+                    timeOutSliceInMillis, useParallel, useTnir, isTnirFirstAttempt, timeOutsliceInMillisForFullTimeout);
```

Comment:
I know this might be a readability issue, but possible to format well and change this to a ternary since it's an if condition attempting to assign a variable.

```java
InetSocketAddress inetSocketAddress = (0 == timeOutFullInSeconds) ? tdsChannel.open(serverInfo.getServerName(), serverInfo.getPortNumber(), 0, useParallel,useTnir, isTnirFirstAttempt, timeOutsliceInMillisForFullTimeout);
                                                                  : tdsChannel.open(serverInfo.getServerName(), serverInfo.getPortNumber(), timeOutSliceInMillis, useParallel, useTnir, isTnirFirstAttempt, timeOutsliceInMillisForFullTimeout);

```

Might be even better to use an if statement to set timeOutSliceInMillis to 0 if timeOutFullInSeconds since it's the only parameter changed.

```java
int effectiveTimeoutValue = (0 == timeOutFullInSeconds) ? 0 : timeOutSliceInMillis;
InetSocketAddress inetSocketAddress = tdsChannel.open(serverInfo.getServerName(), serverInfo.getPortNumber(), effectiveTimeoutValue, useParallel,useTnir, isTnirFirstAttempt, timeOutsliceInMillisForFullTimeout);
```

Reply:
hmm I think more clear if we just do this since that's the only thing that changed:

inetSocketAddress = tdsChannel.open(serverInfo.getServerName(), serverInfo.getPortNumber(), **(0 == timeOutFillInSeconds) ? 0 : timeOutSliceInMillis**,  useParallel, useTnir, isTnirFirstAttempt, timeOutsliceInMillisForFullTimeout);